v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -495,6 +495,33 @@ func (tc *KubePerTestContext) PatchResourceAndWaitForState(
 	tc.Eventually(new).Should(tc.Match.BeInState(status, severity, gen))
 }
 
+func (tc *KubePerTestContext) CreateSecret(
```

Comment:
Wondering if this should be called `CreateSimpleSecret`, and should have method comments explaining that if data isn't specified it'll be autogenerated?

Reply:
Though TBH I think you could maybe just drop this bit:
```
	if secretData == "" {
		secretData = tc.Namer.GeneratePasswordOfLength(40)
	}
```

because even with this the usage is 

`tc.CreateSecret("mysecret", "mykey", "")` which I am not sure is clearer than `tc.CreateSecret("mysecret", "mykey", tc.GeneratePasswordOfLength(40)` and even though the second _is_ a bit longer it's also (IMO) clearer about what's happening compared to the first one.